internal/app/summon/command/temp_factory.go

```diff
@@ -4,12 +4,13 @@ import (
 	"io/ioutil"
 	"os"
 	"strings"
-
-	"github.com/mitchellh/go-homedir"
 )
 
-// DEVSHM is the location of a memory-mapped directory on Linux.
-const DEVSHM = "/dev/shm"
+// This is the location of a memory-mapped directory on Linux. We want
+// to use this location where possible since files saved there are not
+// saved on disk but kept in transient memory which is more secure than
+// the filesystem.
+const defaultSharedMemoryDir = "/dev/shm"
```

Comment:
I get where you're coming from and we could clean this up at some point (I didn't write any of the original code for it) though to be fair it's probably of dubious value compared to all the other stuff we have to do imo. Ideally, this class wouldn't really care about `/dev/shm` since `/tmp` is almost always on `tmpfs` anyways. The module is also there not to make sure stuff secure but to ensure cleanup after temp fles are used from what I can tell.

> Anyway, you are right that technically it's not in scope, but in general I like to clean things up whenever I'm there anyway

This is the big issue to me with this PR: originally the only change needed in this file was a _single line of logic_ that was then: expanded to add tests since it had none (my own choice there), that was then expanded to rename shm variable, that was then expanded to add a secondary constructor, that then could have been cleaned up more, etc, etc which really isn't the way that we should work with PRs for issues that are unrelated to the task at hand especially with long CR cycles. Incremental fixes are fine but we can't make everything we touch perfect first time around.

Reply:
> to be fair it's probably of dubious value compared to all the other stuff we have to do imo.

I agree.  I don't care so much about this particular file and making sure that we keep our quality bar really high for new code or updates to code.

> This is the big issue to me with this PR: originally the only change needed in this file was a single line of logic that was then: expanded to add tests since it had none (my own choice there), that was then expanded to rename shm variable, that was then expanded to add a secondary constructor, that then could have been cleaned up more, etc, etc which really isn't the way that we should work with PRs for issues that are unrelated to the task at hand especially with long CR cycles. Incremental fixes are fine but we can't make everything we touch perfect first time around.

I agree here too.  I wasn't aware of all that.  My rule is: If you're there doing something, and notice something that should be fixed that will take anywhere from 15m - 2hrs or so (the higher times being justified only for bigger problems), then just do it.  If it's bigger than that, create an issue.

It sounds like you had already gone beyond the call of duty, and then added another beyond the call of duty on top of that, in this particular case.  So I think we're on the same page.